rules/rbe_repo.bzl

```diff
@@ -63,65 +102,154 @@ Add to your WORKSPACE file the following:
+toolchain_config_suite_spec. It will generate configs for the current version
+of bazel you are running with (overridable via attr).
+This will also (abusing Bazel hermeticity principles) modify the versions.bzl
+file in the output_base. This is so that subsequent executions of the rule
+(by you, or by any of your users after you have checked-in these generated files)
+will be able to directly use them without having to generate them again.
+
+For users of your toolchain_config_suite_spec all that they need to do is
+add to their WORKSPACE:
+
+  <Add the bazel-toolchains repo http_archive, same as above>
+
+  load("//path/to_your/toolchain_config_suite_spec.bzl", "your_toolchain_config_suite_spec_stuct")
+
+  rbe_autoconfig(
+    name = "rbe_your_custom_toolchain_config_suite_spec",
+    toolchain_config_suite_spec = your_toolchain_config_suite_spec_stuct,
+  )
+
+And that's it! They should be able to get checked-in configs every time,
+as long as:
+  - Whenever there is a new Bazel version that is needed for RBE builds,
+    you, the owner of the toolchain_config_suite_spec generates and
+    publishes the new toolchain configs to your repo (this is because
+    new Bazel versions can only be guaranteed to work with toolchain
+    configs that were generated for the specific version of Bazel used)
+  - Whenever there is a new Bazel version that is needed for RBE builds,
+    the users of your toolchain_config_suite_spec update their pin to
+    your repo.
+
+If you want to create (more) different sets of toolchain configurations (a toolchain_config_spec)
+with a different set of env variables, you can do so by reusing the
+toolchain_config_suite_spec, and providing a distinct toolchain_config_spec_name. Example:
+
+rbe_autoconfig(
+    name = "rbe_custom_env2",
+    env = {<dict declaring env variables>},
+    export_configs = True,
+    toolchain_config_spec_name = "<unique name to assign this toolchain_config_spec>",
+    toolchain_config_suite_spec = your_toolchain_config_suite_spec_stuct,
+)
+
+rbe_autoconfig(
+    name = "rbe_custom_env2",
+    env = {<dict declaring env variables>},
+    export_configs = True,
+    toolchain_config_spec_name = "<unique name to assign this toolchain_config_spec>",
+    toolchain_config_suite_spec = your_toolchain_config_suite_spec_stuct,
+)
+
+NOTES:
+
+READ CAREFULLY THROUGH THESE NOTES, NO MATTER YOUR USE CASE:
+
+NOTE 1: SETTING TOOLCHAIN FLAGS
+
+This is not an up to date source for flags, and just provides general guidance
+please see //bazelrc/.latest.bazelrc for the most up to date flags.
+
+Once you have added the rbe_autoconfig rule to your WORKSPACE, you will
+need to set up toolchain flags that select the appropriate toolchain configs.
+The flags below, show an sample of those flags, which was last reviewed
+with Bazel 0.25.0 and for a rbe_autoconfig rule with name 'rbe_default'.
+If you are using a later version of Bazel or your target has a different
```

Comment:
Replace `target` with `rbe_autoconfig rule target` to clarify which target you are referring to here

Reply:
done